src/ZfrCors/Factory/CorsServiceFactory.php

```diff
@@ -35,11 +36,20 @@ class CorsServiceFactory implements FactoryInterface
      * {@inheritDoc}
      * @return CorsService
      */
-    public function createService(ServiceLocatorInterface $serviceLocator)
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
     {
         /* @var $corsOptions CorsOptions */
-        $corsOptions = $serviceLocator->get('ZfrCors\Options\CorsOptions');
+        $corsOptions = $container->get('ZfrCors\Options\CorsOptions');
 
         return new CorsService($corsOptions);
     }
+    /**
+     * {@inheritDoc}
+     *
+     * @return CorsService
+     */
+    public function createService(ServiceLocatorInterface $container)
```

Comment:
@macnibblet maybe I don't get what you want because I already added `__invoke` to support zend-servicemanager v3, but the second part where you suggesting remove `Factorinterface` and `createService` this is BC break.

I don't understand what changes you want I think I did everything mentioned here: http://zendframework.github.io/zend-servicemanager/migration/#factoryinterface